service/src/main.rs

```diff
@@ -694,14 +716,69 @@ fn print_events(events: Events, _sender: Sender<String>) {
 	}
 }
 
+#[cfg(feature = "dcap")]
+fn fetch_marblerun_events_every_hour<E>(
+	api: ParentchainApi,
+	enclave: Arc<E>,
+	accountid: AccountId32,
+	is_development_mode: bool,
+	url: String,
+	marblerun_base_url: String,
+) where
+	E: RemoteAttestation + Clone + Sync + Send + 'static,
+{
+	let enclave = enclave.clone();
+	let handle = thread::spawn(move || {
+		const POLL_INTERVAL_1_HOUR_IN_SECS: u64 = 1 * 30;
+		loop {
+			info!("Polling marblerun events for quotes to register");
+			register_quotes_from_marblerun(
+				&api,
+				enclave.clone(),
+				&accountid,
+				is_development_mode,
+				url.clone(),
+				marblerun_base_url.clone(),
+			);
+
+			thread::sleep(Duration::from_secs(POLL_INTERVAL_1_HOUR_IN_SECS));
+		}
+	});
+
+	handle.join().unwrap()
+}
+#[cfg(feature = "dcap")]
+fn register_quotes_from_marblerun(
+	api: &ParentchainApi,
+	enclave: Arc<dyn RemoteAttestation>,
+	accountid: &AccountId32,
+	is_development_mode: bool,
+	url: String,
+	marblerun_base_url: String,
+) {
+	let enclave = enclave.as_ref();
+	let events = prometheus_metrics::fetch_marblerun_events(&marblerun_base_url).unwrap();
+	let quotes: Vec<&[u8]> =
+		events.iter().map(|event| event.get_quote_without_prepended_bytes()).collect();
+
+	for quote in quotes {
+		let ext = enclave
+			.generate_dcap_ra_extrinsic_from_quote(url.clone(), &quote)
+			.expect("Extracting information from valid quotes should never fail; qed");
```

Comment:
I have been thinking about this, and I would like to not have an expect here because you never know if something on the enclave side failed for whatever reason (e.g. bad thread management, which made the ecall fail because the secure thread pool is exhausted). Let's assume that this happens, I would like that the loop here does not abort, but just logs whatever error is returned, and we hope that we will succeed the next time.